templates/repo/diff/blob_excerpt.tmpl

```diff
@@ -4,35 +4,35 @@
 		{{if eq .GetType 4}}
 			<td class="lines-num lines-num-old" data-line-num="{{if $line.LeftIdx}}{{$line.LeftIdx}}{{end}}">
 				{{if or (eq $line.GetExpandDirection 3) (eq $line.GetExpandDirection 5)}}
-					<a role="button" class="blob-excerpt" data-url="{{$.RepoLink}}/blob_excerpt/{{PathEscape $.AfterCommitID}}" data-query="{{$line.GetBlobExcerptQuery}}&style=split&direction=down&wiki={{$.PageIsWiki}}" data-anchor="{{$.Anchor}}">
+					<button class="blob-excerpt button button-ghost" data-url="{{$.RepoLink}}/blob_excerpt/{{PathEscape $.AfterCommitID}}" data-query="{{$line.GetBlobExcerptQuery}}&style=split&direction=down&wiki={{$.PageIsWiki}}" data-anchor="{{$.Anchor}}">
```

Comment:
There is a `a.blob-excerpt` style that no longer matches and this mismatch results in the button no longer taking up full height:

<img width="122" alt="Screenshot 2023-05-20 at 00 41 12" src="https://github.com/go-gitea/gitea/assets/115237/9d2397a1-582f-41e4-bfe0-deb0f61b4d8f">

I think you can likely either change the selector to `button.blob-excerpt` or maybe just `.blob-excerpt`, depending on specificity of other applied styles.

Reply:
Have changed it now to `.blob-excerpt`.

Reply:
Hover does not work correctly:

<img width="117" alt="Screenshot 2023-05-20 at 11 28 40" src="https://github.com/go-gitea/gitea/assets/115237/3e0fd783-7d5f-4c51-bc38-a05a0f6018e0">

Reply:
It is because this is a misuse of `button-ghost`. You can't just plaster that class everywhere and expect it to work.

Suggestion would be to remove it and style only through this special class as these buttons are special to some extend and don't fit any our regular button schemes.

<img width="516" alt="image" src="https://github.com/go-gitea/gitea/assets/115237/cf091ff4-943e-470d-af08-4807c894f1da">
<img width="513" alt="image" src="https://github.com/go-gitea/gitea/assets/115237/a03e4883-cf0d-41cc-87a3-50a644740989">

Reply:
Why does it work for me? 
![image](https://github.com/go-gitea/gitea/assets/51889757/4ea9f3e4-4310-4fb3-9791-2f2bcbe5a46b)

Reply:
Or about what exactly are you talking?

Reply:
Just diff view, viewing files:

<img width="206" alt="Screenshot 2023-05-20 at 11 43 50" src="https://github.com/go-gitea/gitea/assets/115237/4edc3a47-8a84-4a3c-94ee-ebd6af75da69">

I think they do look better with background filling up, like GitHub's:

<img width="147" alt="image" src="https://github.com/go-gitea/gitea/assets/115237/e56e373c-3a6b-49a6-a1df-3f2f1060f4c7">

Reply:
Interesting, I still can't reproduce.
If you know how to fix it, feel free to push a commit.

Reply:
And I agree with the background, although I don't know if that's in scope for this PR or not.

Reply:
I will try. The background effect was there since a long time, it's not a new addtion.

Reply:
Note there is also a `td.blob-excerpt` so `.blob-excerpt` selector alone was not matching exactly. I think I will rename the button to something more sensible like `expander-button`.

Reply:
Pushed https://github.com/go-gitea/gitea/pull/24817/commits/306767b8a5a5db5c62bdb3c48b85e74e18fc2162, please try.

Reply:
See https://github.com/go-gitea/gitea/pull/24817#issuecomment-1555879288

Reply:
Oh, also the buttons don't work anymore.

Reply:
@silverwind for some reason your change doesn't work with firefox.
I always get the state I've screenshoted now, and the buttons don't work anymore.
Did you perhaps not push something?